File: scripts/fix_dfm_resource.py
import urllib.request
import json
import os

SUPABASE_URL = "https://dlflpvmdzkeouhgqwqba.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRsZmxwdm1kemtlb3VoZ3F3cWJhIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2NzcyNzYwOSwiZXhwIjoyMDgzMzAzNjA5fQ.QT2RDwGP92JhDFb3fGRgMuViKW-AioTIu44x_g0hw5o"
headers = {
    "apikey": SUPABASE_KEY,
    "Authorization": f"Bearer {SUPABASE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=representation"
}

req = urllib.request.Request(f"{SUPABASE_URL}/rest/v1/resources?title=ilike.*Checklist%20DFM*", headers=headers)
with urllib.request.urlopen(req) as res:
    data = json.loads(res.read().decode('utf-8'))
    for row in data:
        print(f"ID: {row['id']}, Title: {row['title']}")
        content = row.get("content", "")
        if "\\n" in content:
            print("Found literal \\n. Fixing...")
            new_content = content.replace("\\n", "\n")
            
            patch_req = urllib.request.Request(f"{SUPABASE_URL}/rest/v1/resources?id=eq.{row['id']}", data=json.dumps({"content": new_content}).encode('utf-8'), headers=headers, method="PATCH")
            with urllib.request.urlopen(patch_req) as patch_res:
                print(f"Update response code: {patch_res.getcode()}")
        else:
            print("No literal \\n found.")

File: scripts/fix_dfm_resource.mjs
import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = "https://dlflpvmdzkeouhgqwqba.supabase.co";
const SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRsZmxwdm1kemtlb3VoZ3F3cWJhIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2NzcyNzYwOSwiZXhwIjoyMDgzMzAzNjA5fQ.QT2RDwGP92JhDFb3fGRgMuViKW-AioTIu44x_g0hw5o";

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);

async function run() {
    console.log("Fetching resource...");
    const { data, error } = await supabase
        .from("resources")
        .select("id, title, content")
        .ilike("title", "%Checklist DFM%")
        .single();

    if (error) {
        console.error("Error:", error);
    } else {
        console.log("Title:", data.title);
        console.log("Content includes \\n?", data.content.includes("\\n"));
        console.log("Raw content sample:", JSON.stringify(data.content.substring(0, 100)));
        
        if (data.content.includes("\\n")) {
            console.log("Fixing content...");
            const fixedContent = data.content.replace(/\\n/g, "\n");
            const { error: updateError } = await supabase
                .from("resources")
                .update({ content: fixedContent })
                .eq("id", data.id);
            if (updateError) console.error("Update Error:", updateError);
            else console.log("Fixed successfully in DB.");
        }
    }
}

run();
